### PR TITLE
Bump to Gradle 7 and use Open JDK 11

### PR DESCRIPTION
**Pull request by blasten**
Migrate a single test to Gradle 7 and use Open JDK 11.

Issue: https://github.com/flutter/flutter/issues/87649
Related PR: https://github.com/flutter/flutter/pull/87761

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
LGTM

File: .ci.yaml [hunk 1]
```diff
@@ -1266,41 +1266,61 @@ targets:
...
     presubmit: false
+    bringup: true
     timeout: 60
     properties:
       tags: >
         ["devicelab"]
       task_name: complex_layout_android__compile
+      dependencies: >-
+        [
+          {"dependency": "open_jdk", "version": "11"}
```
Comment (CaseyHillers) [inline .ci.yaml:1277]: You'll also need to override the cache.

```yaml
caches: >-
  [
          {"name":"openjdk","path":"java11"},
  ]
```

**Review by CaseyHillers — CHANGES_REQUESTED** [review]

File: .ci.yaml (hunk 1 above)
Comment (blasten) [reply .ci.yaml:1277]: does it make sense to cover this in the docs too ? https://github.com/CaseyHillers/cocoon/blob/1f1fa81e2588904908d613111a9d72e784cc6e24/CI_YAML.md#upgrading-dependencies

Comment (CaseyHillers) [reply .ci.yaml:1277]: Yes, i'll send a PR

**Review by CaseyHillers — APPROVED** [review]

**Pull request merged**